README.md

```diff
@@ -146,6 +146,140 @@ Module blocks could be transpiled to either data URLs, or to a module in a separ
+
+### Can Module Blocks help with bundling?
+
+Mostly no. In _some_ scenarios Module Blocks can be sufficient to bundle:
+
+```js
+const count = module {
+  let i = 0;
+
+  export function count() {
+    i++;
+    return i;
+  }
+};
+
+const uppercase = module {
+  export function uppercase(string) {
+    return string.toUpperCase();
+  }
+};
+
+const { count } = await import(count);
+const { uppercase } = await import(uppercase);
+
+console.log(count()); // 1
+console.log(uppercase("daniel")); // "DANIEL"
+```
+
+In the _general_ case, however, modules need to refer to each other. For that to work Module Blocks would need to be able to close over modules, which they can’t do:
+
+```js
+const count = module {
+  let i = 0;
+
+  export function count() {
+    i++;
+    return i;
+  }
+};
+
+const uppercase = module {
+  export function uppercase(string) {
+    return string.toUpperCase();
+  }
+};
+
+const combined = module {
+  const { count } = await import(count);
+  const { uppercase } = await import(uppercase);
+
+  console.log(count()); // 1
+  console.log(uppercase("daniel")); // "DANIEL"
+};
+
+// ReferenceError as we can't close over count or uppercase!!
+```
+
+To address the bundling problem, Dan Ehrenberg is maintaining a separate [proposal/idea][js module bundles].
+
+### What about Blöcks?
+
+[Blöcks] has been archived. Module blocks is probably a better fit for JavaScript for a bunch of reasons:
+- Blöcks was trying to introduce a new type of function. Both imply that you can close over/capture values outside that scope. We tried to allow that in Blöcks (because it is expected) which turned out to be a can of worms.
+- Instead, Modules are well-explored, well-specified and well-understood by tooling, engines and developers. A lot of questions we had to worry about in Blöcks are naturally resolved through prior work in the modules space (e.g a module can only reference the global scope and do imports).
+- Modules already have a caching mechanism.
+- Modules are easier to  backwards compatible by making module blocks ObjectURL-able.
+
+### What about TypeScript?
+
+Within the same realm, typings should work in a straightforward way, similarly to how dynamic imports are currently typed.
+
+Across realms, the story is a bit messy, but not due to Module Blocks. It is notoriously difficult to define what kind of scope a TypeScript file should be executed in (Main thread vs worker vs service worker), which is often solved by having multiple tsconfig.json file and composited projects. In that scenario, it’s even harder to have code that is shared across these TS projects.
+
+When communicating with a Worker, you already need to force a type on the `event.data` to bring typing to the communication channel.
+
+All in all, the Module Blocks does not make the typing situation worse or more complicated than it already is.
+
```

Comment:
Should we also clarify how caching works in the FAQ, and what a module block *is* (just a new object)?